index.js

```diff
@@ -48,6 +55,201 @@ class ExecutorQueue extends Executor {
+            winston.info(`*** checked for worker status: ${verb} (event loop delay: ${delay}ms)`));
+
+        this.scheduler.on('start', () =>
+            winston.info('scheduler started'));
+        this.scheduler.on('end', () =>
+            winston.info('scheduler ended'));
+        this.scheduler.on('master', state =>
+            winston.info(`scheduler became master ${state}`));
+        this.scheduler.on('error', error =>
+            winston.info(`scheduler error >> ${error}`));
+        this.scheduler.on('working_timestamp', timestamp =>
+            winston.info(`scheduler working timestamp ${timestamp}`));
+        this.scheduler.on('transferred_job', (timestamp, job) =>
+            winston.info(`scheduler enqueuing job timestamp  >>  ${JSON.stringify(job)}`));
+
+        process.on('SIGTERM', () => {
+            this.multiWorker.end((error) => {
+                if (error) {
+                    winston.error(`failed to end the worker: ${error}`);
+                }
+
+                this.scheduler.end((err) => {
+                    if (err) {
+                        winston.error(`failed to end the scheduler: ${err}`);
+                        process.exit(128);
+                    }
+                    process.exit(0);
+                });
+            });
+        });
+    }
+
+    /**
+     * Posts a new build event to the API
+     * @method postBuildEvent
+     * @param {Object} config          Configuration
+     * @param {Object} config.pipeline Pipeline of the job
+     * @param {Object} config.job      Job object to create periodic builds for
+     * @return {Promise}
+     */
+    async postBuildEvent({ pipeline, job }) {
+        const jwt = this.tokenGen(pipeline.admins[0], {}, pipeline.scmContext);
+
+        const options = {
+            url: '/events',
+            method: 'POST',
+            headers: {
+                Authorization: `Bearer ${jwt}`,
+                'Content-Type': 'application/json'
+            },
+            body: {
+                pipelineId: pipeline.id,
+                startFrom: job.name
+            }
+        };
+
+        return req(options, (err, response) => {
+            if (!err && response.statusCode === 200) {
+                return Promise.resolve(response);
+            }
+
+            return Promise.reject(err);
+        });
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object}   config             Configuration
+     * @param {Object}   config.pipeline    Pipeline of the job
+     * @param {Object}   config.job         Job object to create periodic builds for
+     * @param {Function} config.tokenGen    Function to generate JWT from username, scope and scmContext
+     * @param {Boolean}  config.isUpdate    Boolean to determine if updating existing periodic build
+     * @return {Promise}
+     */
+    async _startPeriodic(config, triggerBuild = false) {
```

Comment:
`triggerBuild` is not invoked by the public function in https://github.com/screwdriver-cd/executor-base. Also this parameter needs to be documented.

Reply:
We can put `triggerBuild` in the config.

Reply:
I've put it in the config for now so it makes it simpler across the board